cockatrice/src/player.cpp

```diff
@@ -2552,7 +2558,7 @@ void Player::updateCardMenu(const CardItem *card)
     bool writeableCard = getLocal();
     if (card->getZone() && card->getZone()->getIsView())
     {
-        ZoneViewZone *view = static_cast<ZoneViewZone *>(card->getZone());
+        auto *view = dynamic_cast<ZoneViewZone *>(card->getZone());
```

Comment:
Is there a reason for this change?

Reply:
I was requested to by CLion as this isn't a static change apparently

Reply:
`CardItem::getZone()` returns a `CardZone*`, and `ZoneViewZone` inherits from`SelectZone` that inherits from `CardZone`. To be clear:
```cpp
class CardItem {
  CardZone *getZone() const { return zone; }
 ...
};
class SelectZone : public CardZone { ... };
class ZoneViewZone : public SelectZone { ... };

```

This is a "cast down" and should work fine with `static_cast` as long as nothing `virtual` is involved.
Using `static_cast`, if we ever make a change that breaks this behaviour, the compiler will shout out an error.
With `dynamic_cast` instead, we won't notice anything until crash logs starts to appear because `dynamic_cast` is failing and returning `nullptr` instead of a valid value, and we have no check in place.

Reply:
I'll keep this in mind for future changes